js/src/features/human-resouce/components/Absence/Abcence.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useCallback } from "react";
+import { List, Tag, Tooltip } from "antd";
+import {
+  CheckCircleOutlined,
+  CloseCircleOutlined,
+  EditOutlined,
+} from "@ant-design/icons";
+import { Absence as AbsenceModel } from "HumanResourceModels";
+
+interface Props {
+  absence: AbsenceModel;
+  canEdit: boolean;
+  approve: (id: string) => any;
+  remove: (id: string) => any;
+  openUpdateModal: (id: string) => any;
+}
+
+export const Absence: React.FC<Props> = ({
+  absence,
+  canEdit,
+  approve,
+  remove,
+  openUpdateModal,
+}) => {
+  const handleApprove = useCallback(() => {
+    if (!absence.isApproved) {
+      approve(absence.id);
+    }
+  }, [absence, approve]);
+
+  const handleRemove = useCallback(() => {
+    remove(absence.id);
+  }, [absence, remove]);
+
+  const handleOpenUpdate = useCallback(() => {
+    openUpdateModal(absence.id);
+  }, [absence, openUpdateModal]);
+
+  const title = `${absence.startedAt} ${absence.endedAt}`;
+  const description = `${absence.reason}: ${absence.comment}`;
+  const content = absence.isApproved ? (
```

Comment:
You can create separated component, e.g. AbsenceStatus. It makes your component more cleaner. Don't shy to make this component in the same file.